src/translation/translator/translator.py

```diff
@@ -0,0 +1,153 @@
+        translation_queue = [ ]
+        key_mapping = { }
+
+        def is_translatable_value (text, key):
+            EMAIL_REGEX = "[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+"
+            URL_REGEX = "(https?://)?\w+(\.\w+)+"
+            UUID_REGEX = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+            DATE_REGEX = "'\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z"
+            SHORT_DATE_REGEX = "\d{4}-\d{2}-\d{2}"
+            DEFAULT_IGNORE_LIST = [
+                "null",
+                "undefined",
+                "None",
+                "True",
+                "False",
+                "null",
+                "false",
+                "true",
+                # let's add some common JSON keyword that should not be translated
+            ]
+
+            if ignore_keys and any(fnmatch.fnmatch(key, pattern) for pattern in ignore_keys):
+                return False
+            elif ignore_patterns and any(re.fullmatch(pattern, text) for pattern in ignore_patterns):
+                return False
+
+            return not (re.fullmatch(EMAIL_REGEX, text) or
+                        re.fullmatch(URL_REGEX, text) or
+                        re.fullmatch(UUID_REGEX, text) or
+                        re.fullmatch(DATE_REGEX, text) or
+                        re.fullmatch(SHORT_DATE_REGEX, text) or
+                        text in DEFAULT_IGNORE_LIST or
+                        text.isspace())
+
+        def collect_translatable (json_struct: dict, current_key = ""):
+            if isinstance(json_struct, dict):
+                for key in json_struct:
+                    new_key = current_key + "." + key if current_key else key
+                    json_struct[ key ] = collect_translatable(json_struct[ key ], new_key)
+
+            elif isinstance(json_struct, list):
+                for idx, item in enumerate(json_struct):
+                    new_key = current_key + "." + str(idx)
+                    json_struct[ idx ] = collect_translatable(item, new_key)
+
+            elif isinstance(json_struct, str) and is_translatable_value(json_struct, current_key):
+                translation_queue.append(json_struct)
+                if json_struct not in key_mapping:
+                    key_mapping[ json_struct ] = [ ]
+                key_mapping[ json_struct ].append(current_key)
+
+            return json_struct
+
+        collect_translatable(json_dict)
+
+        for i in range(0, len(translation_queue), batch_size):
+            translations = self.translate(translation_queue[ i:i + batch_size ], target_language = target_language)
+
+            for translation in translations:
+                translated_text = translation[ 'translatedText' ]
+                original_text = translation[ 'input' ]
+
+                for key_path in key_mapping[ original_text ]:
+                    keys = key_path.split('.')
+                    item_to_replace = json_dict
```

Comment:
Take a look at this: https://github.com/massenergize/api/blob/main/src/_main_/utils/translation/__init__.py#L9

It has almost everything you need including batching of the stuff.